### ISSUE TITLE
Build failing for iOS Simulator

### ISSUE THREAD

**Issue by Steve-0628** [googlevr/cardboard#294]
The SDK in the xcworkspace has problem importing files, and linking the object file.
I was able to fix the import error by adding `$(PROJECT_DIR)/** $(PROJECT_DIR)/../proto` to User Header Search Paths on the SDK, but it caused a link error: 
```ld: in /Users/USERNAME/Library/Developer/Xcode/DerivedData/Cardboard-ghbexfkjdvhfijdxyeolivviadjd/Build/Products/Debug-iphonesimulator/cardboard_api.a(message_lite.o), building for iOS Simulator, but linking in object file built for macOS, for architecture x86_64```
Object linking problem only occurred when I built for iOS Simulator, and building for my iPhone worked.

**Comment by Mandelbrow** [comment]
Have you tried the suggestion from Stackoverflow?

_1.    reset the simulator and retry
2.    if it still doesn't work, remove the app on simulator and re-install it via Xcode debugging_

[https://stackoverflow.com/questions/13105524/ios-app-works-on-device-but-not-on-simulator-anymore](url)

**Comment by Steve-0628** [comment]
Thanks for the reply,
Link error occurs before the simulator start, so I don't think it is simulator related. I also tried building clean, with a simulator that I've never used, but it also failed.

**Comment by jballoffet** [comment]
Thanks @Steve-0628 for reaching out to us! I confirm that I'm able to reproduce the issue. Based on some quick research I did, it seems that the protobuf prebuilt binaries included within the Cardboard SDK (https://github.com/googlevr/cardboard/blob/master/third_party/libprotobuf/ios/lib/libprotobuf-lite.a) are no longer suitable to be used with the iOS Simulator on Xcode 12 onwards. Therefore, you'll need to build them from source (see https://developers.google.com/cardboard/develop/ios/protobuf for guidance).

This issue has been added to our backlog and will be prioritized along with other issues. We'll update this ticket once we have some news about it. Thanks!
